src/components/EditImageModal/index.jsx

```diff
@@ -98,12 +101,22 @@ export default class EditImageModal extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const newState = {};
-    switch (this.props.assetsStatus.type) {
+    switch (nextProps.assetsStatus.type) {
       case assetActions.upload.UPLOAD_ASSET_SUCCESS: {
-        const uploadedAsset = this.props.assetsStatus.response.asset;
-        this.props.selectAsset(uploadedAsset, 0);
+        const uploadedAsset = nextProps.assetsStatus.response.asset;
```

Comment:
I think we should also set `this.state.isStatusAlertOpen` to false and `this.state.uploadErrorMessageBody` to some default state here. If I try to upload an invalid image on page 1, close the status alert, successfully upload an image, it brings me to page 2, where I can see the status alert, which is confusing. I can also see the uploadErrorMessageBody in the state. Is the right place to respond this way to a successful upload?

![image](https://user-images.githubusercontent.com/11871801/40075340-1737938c-584a-11e8-8d60-23cc89f66a16.png)

Reply:
I actually think that we should do that on any movement between pages, because if I click the back button from the above screenshot, I actually see the original error from the previous page (when I tried to originally upload an invalid image), despite the fact that I already have an image selected.

![image](https://user-images.githubusercontent.com/11871801/40075515-785f6428-584a-11e8-8d27-a078cf0f509c.png)

I think clearing the isStatusAlertOpen and uploadErrorMessageBody state on any page change or upload success would take care of that.

Reply:
👍totally agree, I manually tested clearing that on the next button click, but forgot to validate with the automatic movement from the successful upload.